cmd/notation/key.go

```diff
@@ -294,26 +225,19 @@ func deleteKeys(ctx context.Context, opts *keyDeleteOpts) error {
 	logger := log.GetLogger(ctx)
 
 	// core process
-	signingKeys, err := configutil.LoadSigningkeysOnce()
-	if err != nil {
-		return err
-	}
-
-	prevDefault := signingKeys.Default
-	var deletedNames []string
-	for _, name := range opts.names {
-		idx := slices.Index(signingKeys.Keys, name)
-		if idx < 0 {
-			logger.Warnf("Key %s not found, command did not take effect", name)
-			return errors.New(name + ": not found")
-		}
-		signingKeys.Keys = slices.Delete(signingKeys.Keys, idx)
-		deletedNames = append(deletedNames, name)
-		if prevDefault == name {
-			signingKeys.Default = ""
+	var deletedNames []string 
+	var prevDefault string
+	exec := func(s *config.SigningKeys) error {
+		prevDefault = *s.Default
+		var err error
+		deletedNames, err = s.Remove(opts.names)
+		if err != nil {
+			logger.Error("something went wrong while deleting keys, error: %v", err)
```

Comment:
Maybe we could choose a more formal wording here? something like:
```suggestion
			logger.Error("Keys deletion failed to complete with error: %v", err)
```
(Btw, we'd capitalize the log messages.)